File: tests/pages/offboarding_page.py
"""
Offboarding & Separation Workspace Page Object (/offboarding).
"""

from playwright.sync_api import Page, expect
from tests.pages.base_page import BasePage
from tests.components.modal import ModalComponent


class OffboardingPage(BasePage):
    PATH = "/offboarding"

    def __init__(self, page: Page):
        super().__init__(page)
        self.submit_resignation_btn = (
            page.locator("[data-testid='submit-resignation-btn'], button")
            .filter(has_text="Submit Resignation")
            .first
        )
        self.modal = ModalComponent(page)
        self.notice_input = page.locator("#noticeDaysInput, input[type='number']").first
        self.resig_date_input = page.locator("#resigDateInput, input[type='date']").first

    def initiate_resignation(self, reason: str = "Pursuing new career opportunity", lwd: str = "2026-09-30"):
        """Submits employee resignation."""
        self.submit_resignation_btn.wait_for(state="visible", timeout=12000)
        self.submit_resignation_btn.click()
        self.page.wait_for_timeout(600)

File: e2e/pages/OffboardingPage.ts
import { Page, Locator, expect } from "@playwright/test";
import { BasePage } from "./BasePage";

export class OffboardingPage extends BasePage {
  readonly employeeSelect: Locator;
  readonly resignationDateInput: Locator;
  readonly noticeDaysInput: Locator;
  readonly submitResignationButton: Locator;
  readonly clearanceCheckboxes: Locator;
  readonly approveFfButton: Locator;

  constructor(page: Page, baseURL?: string) {
    super(page, baseURL);
    this.employeeSelect = page.locator('select#resigEmpSelect, select');
    this.resignationDateInput = page.locator('input#resigDateInput, input[type="date"]').first();
    this.noticeDaysInput = page.locator('input#noticeDaysInput, input[type="number"]').first();
    this.submitResignationButton = page.locator('button[type="submit"]:has-text("Initiate"), button:has-text("Submit Resignation")');
    this.clearanceCheckboxes = page.locator('input[type="checkbox"]');
    this.approveFfButton = page.locator('button:has-text("Approve F&F"), button:has-text("Approve Settlement")');
  }

  async goto(): Promise<void> {
    await this.navigate("/offboarding");
  }

  async assertLoaded(): Promise<void> {
    await expect(this.page).toHaveURL(/.*\/offboarding.*/);
    await expect(this.page.locator("body")).toContainText(/Offboarding|Separation/i);
  }

  async submitResignation(noticeDays: number = 30): Promise<void> {
    await this.noticeDaysInput.fill(String(noticeDays));
    await this.submitResignationButton.click();
  }
}
